### PR TITLE
Resolve Bugs in Github URLs

### PR DESCRIPTION
**Pull request by TG1999**
Signed-off-by: Tushar Goel <tushar.goel.dav@gmail.com>

### CODE REVIEW

**Comment by TG1999** [comment]
@tdruez FYI

File: src/packageurl/contrib/url2purl.py [hunk 1]
```diff
@@ -409,7 +409,16 @@ def build_github_purl(url):
...
+    download_pattern = (
+        r"https?://github.com/downloads/(?P<namespace>.+)"
+        r"/(?P<name>.+)/v?(?P<version>.+).(zip|tar.gz|tar.bz2|.tgz)"
+    )
+
+    git_pattern = (
+        r"https?://github.com/(?P<namespace>.+)"
+        r"/(?P<name>.+).(git)"
+    )
+    for pattern in [archive_pattern, raw_pattern, blob_pattern, releases_download_pattern, download_pattern, git_pattern]:
```
Comment (pombredanne) [inline src/packageurl/contrib/url2purl.py:421]: What about using this?
```suggestion

    patterns = (
        archive_pattern,
        raw_pattern,
        blob_pattern,
        releases_download_pattern,
        download_pattern,
        git_pattern,
    )

    for pattern in patterns:
```

**Review by pombredanne — CHANGES_REQUESTED** [review]
Thank you ++
See my comment inline for your consideration

File: src/packageurl/contrib/url2purl.py (hunk 1 above)
Comment (TG1999) [reply src/packageurl/contrib/url2purl.py:421]: Agreed :)

**Comment by TG1999** [comment]
Done @pombredanne , please give it a look :)

**Comment by tdruez** [comment]
@pombredanne We need your input on this discussing before approving those changes:
https://github.com/package-url/packageurl-python/issues/47#issuecomment-766548093

**Pull request merged**